aten/src/ATen/native/Convolution.cpp

```diff
@@ -696,7 +705,15 @@ at::Tensor _convolution(
     }
 #endif
   } else if (input.device().type() == c10::DeviceType::CPU || input.device().type() == c10::DeviceType::CUDA) {
-    if (params.use_cpu_depthwise3x3_winograd(input, weight)) {
+    if (params.use_xnnpack(input, weight, bias)) {
+      // This is NOT the preferred usage pattern as it forces expensive and
+      // unnecessary weight prepacking per invocation.  Still if we end up here,
+      // which we shouldn't if we have correctly intercepted and replaced this
+      // usage through JIT, still this code path is more efficient for NHWC
+      // tensors than the alternatives.
```

Comment:
I'd suggest dropping this comment or making it much shorter.  This message needs to be sent to the model developers or PyTorch frontend developers, who won't be the ones reading this code.  "Using prepacked conv is preferred, but XNNPACK is still the fastest option for NHWC."

Reply:
Sounds good, will change.